### PR TITLE
[FLINK-33974] Implement the Sink transformation depending on the new SinkV2 interfaces

### PR DESCRIPTION
**Pull request by pvary**
## What is the purpose of the change

Implement the changes to the Sink transformation which should depend only on the new API interfaces. The tests should remain the same, to ensure backward compatibility.

## Brief change log

- SinkTransformationTranslator.java, CommitterOperatorFactory.java, SinkWriterOperator.java, StatefulSinkWriterStateHandler.java - to depend on the new API interfaces instead of the old ones
- Idenpotent change in SinkV2TransformationTranslatorITCase.java - this left out during the test refactor
- StreamGraphGeneratorTest.java, TestExpandingSinkWithMixin.java - Single specific test for the mixin interfaces

## Verifying this change

Added a single test method to the new API.
Kept all of the tests unchanged to make sure that the change is backward compatible.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/SinkTransformationTranslator.java [hunk 1]
```diff
@@ -173,61 +185,76 @@ private void expand() {
             }
         }
 
-        private <CommT> void addCommittingTopology(Sink<T> sink, DataStream<T> inputStream) {
```
Comment (gyfora) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/SinkTransformationTranslator.java:176]: Can we please refert the generic parameter name change? This would avoid some unnecessary changes in the commit

**Review by gyfora — APPROVED** [review]
Aside from a minor comment, the changes look good. New integration test coverage will be introduced as the next PR.  (to make sure we don't lose any coverage / preserve backward compatibility here).

👍

**Comment by pvary** [comment]
@flinkbot run azure
